### ISSUE TITLE
refactor PackageObjectFactoryTest.testCreateModuleWithNonExistName

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#5677]
identified at https://github.com/checkstyle/checkstyle/pull/5637#discussion_r178369217

https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java#L154

```
    @Test
    public void testCreateModuleWithNonExistName() {
        testCreateModuleWithNonExistName("NonExistClassOne");
        testCreateModuleWithNonExistName("NonExistClassTwo");
    }

    private void testCreateModuleWithNonExistName(String name) {
        try {
            factory.createModule(name);
            fail("Exception is expected");
        }
        catch (CheckstyleException ex) {
            final String attemptedNames = BASE_PACKAGE + PACKAGE_SEPARATOR + name + STRING_SEPARATOR
                    + name + CHECK_SUFFIX + STRING_SEPARATOR
                    + BASE_PACKAGE + PACKAGE_SEPARATOR + name + CHECK_SUFFIX;
            final LocalizedMessage exceptionMessage = new LocalizedMessage(0,
                    Definitions.CHECKSTYLE_BUNDLE, UNABLE_TO_INSTANTIATE_EXCEPTION_MESSAGE,
                    new String[] {name, attemptedNames}, null, factory.getClass(), null);
            assertEquals("Invalid exception message",
                    exceptionMessage.getMessage(), ex.getMessage());
        }
}
```

refactor code to have assertEquals in method with annotation `@Test`.
Optimize PMD suppression that is discussed in comment https://github.com/checkstyle/checkstyle/pull/5637#discussion_r178369217

**Comment by pbludov** [comment]
Why should this code be executed twice? Does the factory cache something in case of failure?

**Comment by rnveach** [comment]
@pbludov Doing a blame on the file and looking at the issues/PR will usually tell you everything you need to know.
Reason can be found at: https://github.com/checkstyle/checkstyle/pull/4292#discussion_r113592750
I asked this same question.

**Comment by pbludov** [comment]
I correctly understand that there are not two tests here, but one test and one disguised test setup?

**Comment by romani** [comment]
If such code was introduced to satisfy cobertura coverage, it is most likely not required any more, feel free to make more reasonable for today state of project.

**Comment by rnveach** [comment]
> If such code was introduced to satisfy cobertura coverage

It was to satisfy code coverage itself, not just cobertura. Removal can confirm if coverage goes missing.

> but one test and one disguised test setup?

Like PR stated, one was to initialize the cache, the other was to use the cache. It is to cover both branches of `if (thirdPartyNameToFullModuleName == null)`.

**Comment by romani** [comment]
Asserts should in methods with Test annotation, code need be refactored.

**Comment by romani** [comment]
@surajumang, can you help us to resolve this issue ?

**Comment by surajumang** [comment]
I am on it

**Comment by surajumang** [comment]
@romani how should i proceed with the issue. I am thinking about moving the code from the private method to the method annotated with Test. Please let me know what you think.

**Comment by romani** [comment]
Yes, this what we expect, simple fix. You can keep in private method all required util functionality, but asserts should be in test method.

**Comment by rnveach** [comment]
@romani Do you want to extend this issue for all suppressions listed in `JUnitTestsShouldIncludeAssert` or are you going to create a new issue for each?
